src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -763,32 +807,32 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebooks(int accountId, cons
             QString currDeviceCalendarId = notebook->pluginName().mid(7);
             if (m_serverCalendarIdToCalendarInfo[accountId].contains(currDeviceCalendarId)) {
                 // the server-side calendar exists on the device.
-                // we don't need to purge it, but we may need to update its summary/color details.
-                deviceCalendarIds.append(currDeviceCalendarId);
-                if (notebook->name() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary
-                        || notebook->color() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color
-                        || notebook->description() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description
-                        || notebook->isReadOnly()) {
-                    // calendar information changed server-side.
-                    notebook->setIsReadOnly(false);
-                    notebook->setName(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary);
-                    notebook->setColor(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color);
-                    notebook->setDescription(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description);
-                    m_storage->updateNotebook(notebook);
-                    m_storageNeedsSave = true;
+                if (needCleanSync) {
+                    // we are performing a clean sync cycle.
+                    // we will eventually delete and then insert this notebook.
+                    SOCIALD_LOG_DEBUG("queuing clean sync of local calendar" << notebook->name() << currDeviceCalendarId << "for Google account:" << accountId);
+                    m_serverCalendarIdToCalendarInfo[accountId][currDeviceCalendarId].change = GoogleCalendarSyncAdaptor::CleanSync;
```

Comment:
Right, there are definitely things we can do to reduce the number of clean-syncs which occur as opposed to delta syncs (especially since there are less cases which can cause database corruption now).

But, IMO, unless it proves to be a common occurrence in practice, I think it's a lower priority issue.  The way we currently store sync timestamps (in an out-of-band database) is terrible but it's that way for historical reasons.  At some stage we need to refactor the plugins to get rid fo the remaining "multiple account" isms, and get rid of the false separation between SocialNetworkSyncAdaptor and Buteo::ClientPlugin, so that we can make "proper" use of the Buteo API within our sync adaptors (ie: return the correct error codes, instead of just InternalError; make use of the Buteo sync-anchors timestamp database; make proper use of the Buteo SyncResults struct to inform whether we need to do cleansync or not; etc).

But, not today :-)